src/Arcus.Scripting.SQL/Scripts/RunDatabaseScript.ps1

```diff
@@ -0,0 +1,158 @@
+
+#Get all tables on the database
+$query = 'SELECT TABLE_NAME FROM information_schema.tables'
+$tables = QueryDatabaseWithResult $params $query
+
+# Get script to determine DB version
+$selectDbVersionScript = 'SELECT TOP 1 CurrentVersionNumber FROM DatabaseVersion ORDER BY CurrentVersionNumber DESC'
+
+#Create a DatabaseVersion table if it doesn't exist
+if(!$tables.ItemArray.Contains('DatabaseVersion'))
+{
+    Write-Host 'DatabaseVersion does not exist yet in this database.'
+    Write-Host 'Creating DatabaseVersion Table...'
+    $createDbQuery = GetScriptFileText $scriptpath 'CreateDatabaseVersionTable'
+    QueryDatabase $params $createDbQuery
+    Write-Host 'DatabaseVersion Table Created'
+
+    $databaseVersionNumber = 0
+}
+else
+{
+    # Get database versionnumber
+    Write-Host 'Getting Current Database Version Number...'
+    $databaseVersionNumberDataRow = QueryDatabaseWithResult $params $selectDbVersionScript
+
+    if ($databaseVersionNumberDataRow.ItemArray.Count -eq 0)
+	{
+		$databaseVersionNumber = 0;
+	}
+	else
+	{
+		$databaseVersionNumber = [convert]::ToInt32($databaseVersionNumberDataRow.ItemArray[0])
+	}
+}
+
+
+Write-Host "Current databaseVersionNumber : $databaseVersionNumber"
+
+#Run all necessary scripts
+$files = Get-ChildItem -Path $scriptPath -Filter "ITSME-Enroll_*.sql" | Sort-Object {$_.BaseName -replace "\D+" -as [Int]}
+
+for ($i=0; $i -lt $files.Count; $i++) 
+{
+    $fileName = $files[$i].BaseName
+    $fileNameSections = $fileName.Split('_');
+
+    if ($fileNameSections.Length -lt 2)
+    {
+        Write-Host "File $fileName skipped for not having all required name sections (version and description)."
+        continue;
+    }
+
+    $fileVersionStr = $fileNameSections[1];
+
+    [int] $scriptVersionNumber = -1;
+
+    if (-Not([int32]::TryParse($fileVersionStr, [ref]$scriptVersionNumber)))
+    {
+        Write-Host "File $fileName skipped because version is not valid."
+        continue;
+    }
+
+	[string] $scriptVersionDescription = [convert]::ToString($fileNameSections[2])
+    if($scriptVersionDescription.Length -gt 256)
+    {
+		Write-Host "Need to truncate the migration description because its size is" $scriptVersionDescription.Length "while the maximum size is 256"
+        $scriptVersionDescription = $scriptVersionDescription.Substring(0,255)
+    }
+	
+    Write-Host "Found migration #$scriptVersionNumber with description '$scriptVersionDescription'"
+
+    if($scriptVersionNumber -eq ($databaseVersionNumber + 1))
+	{
+		Write-Host "Running script #$scriptVersionNumber"
+
+        # Perform migration
+		$currentfilepath = "$scriptPath/$fileName.sql"
+		$query = [IO.File]::ReadAllText($currentfilepath)
+		QueryDatabase $params $query
+
+        # Append the new version and description in version table
+        $updateVersionQuery = "INSERT INTO [dbo].[DatabaseVersion] ([CurrentVersionNumber], [MigrationDescription]) VALUES ($scriptVersionNumber, '$scriptVersionDescription')"
```

Comment:
Any possibility to have the database-schame set to a custom value? for instance: 'arcus' instead of 'dbo'?

Reply:
I wonder why you'd do that ?  It's the 'DatabaseVersion' table that is used to keep track of the current version of the DB.  I think dbo is a good start.  Could be an improvement to parameterize that, but I currently don't see any value in that.

Reply:
I do agree that in 99% of the cases the dbo-schema will be used. However, I'm working on a project where we're only granted access rights to a specific schema. So, in our case, this hard-coded schema would make it impossible to use this, which would be a shame 🙂

Reply:
Added a parameter with default `"dbo"` database schema.